File: frontend/src/services/documentService.ts
import { get, post, del } from './api';
import type { Document as DocType } from '../types';

export const documentService = {
  list(params?: Record<string, string | number | boolean | undefined>): Promise<DocType[]> {
    return get<DocType[]>('/documents', params);
  },

  get(id: number): Promise<DocType> {
    return get<DocType>(`/documents/${id}`);
  },

  byPersonne(personneId: number): Promise<DocType[]> {
    return get<DocType[]>('/documents', { personne_id: personneId });
  },

  upload(personneId: number, file: File, typeDocument: string, numero?: string): Promise<DocType> {
    const formData = new FormData();
    formData.append('personne_id', String(personneId));
    formData.append('type_document', typeDocument);
    formData.append('fichier', file);
    if (numero) formData.append('numero', numero);
    return post<DocType>('/documents', formData);
  },

  delete(id: number): Promise<void> {
    return del<void>(`/documents/${id}`);
  },

  downloadUrl(id: number): string {
    const base = (import.meta.env.VITE_API_URL || '/api').replace(/\/+$/, '');
    return `${base}/documents/${id}/download`;
  },

  types(): Promise<Record<string, string>> {
    return get<Record<string, string>>('/documents/types');
  },
};

File: gs_operateur_mobile/lib/services/document_service.dart
import 'dart:io';
import 'package:dio/dio.dart';
import '../models/document.dart';
import 'api_service.dart';

class DocumentService {
  final ApiService _api = ApiService();

  Future<List<Document>> list({String? search, int? personneId, int page = 1}) async {
    final response = await _api.get('/documents', queryParameters: {
      if (search != null && search.isNotEmpty) 'search': search,
      if (personneId != null) 'personne_id': personneId,
      'page': page,
    });

    if (response['success'] == true) {
      final data = response['data'];
      return (data is List ? data : data['data'] ?? [])
          .map<Document>((d) => Document.fromJson(d))
          .toList();
    }
    throw Exception(response['message'] ?? 'Erreur');
  }

  Future<Document> get(int id) async {
    final response = await _api.get('/documents/$id');
    if (response['success'] == true) {
      return Document.fromJson(response['data']);
    }
    throw Exception(response['message'] ?? 'Erreur');
  }

  Future<Document?> upload({
    required int personneId,
    required String typeDocument,
    String? numero,
    required File fichier,
  }) async {
    final formData = FormData.fromMap({
      if (personneId > 0) 'personne_id': personneId,
      'type_document': typeDocument,
      if (numero != null && numero.isNotEmpty) 'numero': numero,
      'fichier': await MultipartFile.fromFile(fichier.path),
    });
    final response = await _api.postFormData('/documents', formData: formData);
    if (response['success'] == true) {
      try {
        return Document.fromJson(response['data']);
      } catch (_) {
        return null;
      }
    }
    throw Exception(response['message'] ?? 'Erreur lors de l\'upload');
  }

  Future<void> delete(int id) async {
    final response = await _api.delete('/documents/$id');
    if (response['success'] != true) {
      throw Exception(response['message'] ?? 'Erreur lors de la suppression');
    }
  }

  String downloadUrl(int id) {
    return '${ApiService().getBaseUrl()}/documents/$id/download';
  }
}
